src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java

```diff
@@ -243,22 +310,52 @@ protected void retrieve(@CheckForNull final SCMSourceCriteria criteria,
             @Override
             public Void run(GitClient client, String remoteName) throws IOException, InterruptedException {
                 final Repository repository = client.getRepository();
-                listener.getLogger().println("Getting remote branches...");
+                listener.getLogger().println("Getting remote branches/tags...");
                 try (RevWalk walk = new RevWalk(repository)) {
                     walk.setRetainBody(false);
-                    for (Branch b : client.getRemoteBranches()) {
+
+                    List<GitObject> gitObjects = new ArrayList<GitObject>();
+                    gitObjects.addAll(client.getRemoteBranches());
+
+                    Hudson hudson = Hudson.getInstance();
+                    DescriptorImpl descriptor = (DescriptorImpl) hudson.getDescriptor(GitSCM.class);
+
+                    if (descriptor.isShowTags()) {
```

Comment:
:bug: fetching tags should be determined by a `TagDiscoveryTrait` not some setting on a global descriptor